app/src/test/java/org/mozilla/fenix/search/toolbar/ToolbarViewTest.kt

```diff
@@ -115,6 +119,31 @@ class ToolbarViewTest {
         verify(exactly = 2) { interactor.onTextChanged(any()) }
     }
 
+    @Test
+    fun `GIVEN search term is set WHEN switching to edit mode THEN the cursor is set at the end of the search term`() {
+        val view = buildToolbarView(false)
+        mockkObject(FeatureFlags)
+        every { testContext.settings().showUnifiedSearchFeature } returns true
+
+        view.update(defaultState.copy(searchTerms = "search terms"))
+
+        // editMode gets called when the view is initialized.
+        verify(exactly = 1) { toolbar.editMode(Toolbar.CursorPlacement.ALL) }
```

Comment:
Looks like this test is failing. I think you need to move this line right after `buildToolbarView`. We can remove the comment afterwards.

Reply:
After the latest push the unified search setting was not set correctly in the test, making it indeed select all the text when switching to edit mode. It should now call `editMode(Toolbar.CursorPlacement.ALL)` only when initializing `ToolbarView`.